lib/core/src/display/rendering/htmlrenderer.rs

```diff
@@ -0,0 +1,63 @@
+use super::Camera;
+use super::HTMLScene;
+use super::IntoCSSMatrix;
+use crate::system::web::StyleSetter;
+use crate::prelude::*;
+
+fn eps(value: f32) -> f32 {
+    if value.abs() < 1e-10 { 0.0 } else { value }
+}
+
+/// A renderer for `HTMLObject`s.
+#[derive(Default)]
+pub struct HTMLRenderer {}
+
+impl HTMLRenderer {
+    /// Creates a HTMLRenderer.
+    pub fn new() -> Self { default() }
+
+    /// Renders the `Scene` from `Camera`'s point of view.
+    pub fn render(&self, camera: &mut Camera, scene: &HTMLScene) {
+        let half_d = scene.get_dimensions() / 2.0;
+        // Note [znear from projection matrix]
+        let expr = camera.projection[(1, 1)]; // expr = 2.0 * near / (height)
+        let near = expr * half_d.y;
+
+        let near = format!("{}px", near);
+        scene.div.element.set_property_or_panic("perspective", &near);
+
+        let mut transform = camera
+                      .transform.to_homogeneous().try_inverse()
+                      .expect("Render couldn't get camera's matrix inverse")
+                      .map(|a| eps(a));
+
+        // Negating the second column to invert Y.
+        // Equivalent to scaling by (1.0, -1.0, 1.0).
+        transform.row_part_mut(1, 4).iter_mut().for_each(|a| *a = -*a);
```

Comment:
instead of writing comment + code which is hard to understand, what about refactoring it to a function with a nice name like `invert_y`? :)